_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+---
+layout: blog_post
+title: "How to Deploy a .NET Container with AWS ECS Fargate"
+author: [deepu-sasidharan, chase-aucoin]
+by: advocate
+communities: [.net]
+description: "This is a tutorial on how to secure your .NET containerized chat app with AWS ECS Fargate and Okta."
+tags: [containers, aws, fargate, c-sharp, csharp, dotnet, dot-net, asp-dot-net, vuejs]
+tweets:
+  - "Ever played with AWS Fargate? Learn how to authenticate users to your #dotnetcore #container hosted in #aws ->."
+  - "Unsure about #containerized #dotnet with #Fargate in #AWS? We've got a primer just waiting for you ->"
+  - "Learn how to create secure .NET #containers in #AWS with Fargate ->"
+image: blog/featured/okta-dotnet-mouse-down.jpg
+type: conversion
+---
+
+In a previous article, we learned [how to host a serverless .NET application using AWS Lambda](/blog/2020/06/08/serverless-lambda-functions-csharp). We talked about the history of serverless and how companies are using these types of technology to simplify delivering APIs and functionality faster than traditional methods. Some problems will arise in this type of application when you need more capability than your standard HTTP requests like GET, POST, PUT, DELETE, etc. Web Sockets is a great example of this.
+
+{% include toc.md %}
+
+## Understanding Containers vs. Virtual Machines
+
+Around 2013, [Docker](https://www.docker.com/) was released, beginning a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes; in application, no.
```

Comment:
```suggestion
In 2013, [Docker](https://www.docker.com/) was released, beginning a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations had been leveraging *virtual machines* with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes; in application, no.
```
preciseness, style guide - avoid capitalizing common nouns, use italics to call out terminology on first use